### PR TITLE
Add hashtree function for returning all buckets.

### PR DESCRIPTION
**Pull request by cmeiklejohn**
Prototype a hashtree function for returning all buckets as an orddict to allow for batching of requests which need to iterate all of the buckets in the tree.

### CODE REVIEW

**Comment by cmeiklejohn** [comment]
cc: @jtuple

File: src/hashtree.erl [hunk 1]
```diff
@@ -479,6 +492,22 @@ esha_final(Ctx) ->
     crypto:sha_final(Ctx).
 -endif.
 
+-spec get_buckets(integer(), orddict(), hashtree()) -> orddict().
+get_buckets(Level, Results, #state{width=W,levels=L}=State) ->
```
Comment (Vagabond) [inline src/hashtree.erl]: So when level=3 that means grab the first three levels, not just the third level?

Also, the math below is a little obtuse, maybe @jtuple can eyeball it.

Comment (cmeiklejohn) [inline src/hashtree.erl]: No, it's just get that level for this function.  The other function generates a sequence up to the provided level, and then folds the results into an orddict.  

See: https://github.com/basho/riak_core/pull/552/files#diff-61653aa3ba5c9c8b67d8e8953ea7e8c4R452

Yes, @jtuple, please check this math.

Comment (Vagabond) [inline src/hashtree.erl]: Oh yes, I forgot that.

**Comment by Vagabond** [comment]
I'm +1 on this, but I want @jtuple to sign off on that tree math before it merges.

**Comment by cmeiklejohn** [comment]
@jtuple thoughts?

File: src/hashtree.erl [hunk 2]
```diff
@@ -479,6 +492,22 @@ esha_final(Ctx) ->
...
+    ByLevel = erlang:trunc(W / (L - 1)),
+    Buckets = case Level of
+        1 ->
+            [0];
+        X ->
+            lists:seq(0, erlang:trunc(ByLevel * (X - 1)))
```
Comment (engelsanchez) [inline src/hashtree.erl]: So ByLevel here is definitely a whole number. If you are worried about Level not being a whole number maybe add a guard instead? If not worried about that, the truncation here is just distracting.

Comment (cmeiklejohn) [inline src/hashtree.erl]: How is it definitely a whole number?

Comment (cmeiklejohn) [inline src/hashtree.erl]: Oh wait, I thought this comment was on the other line.

**Comment by cmeiklejohn** [comment]
Once merged, please merge: basho/riak_kv#863 and basho/riak_repl#557.

**Comment by cmeiklejohn** [comment]
Updated PR with commit https://github.com/basho/riak_core/commit/2c488b7017596df73a9e230116af34bb0ba179a7 which addresses the incorrect math in the original commit.

**Comment by cmeiklejohn** [comment]
Considering we are not merging the `riak_repl` and `riak_kv` changes which require this support in `riak_core`, we're not going to merge this for the 2.0 RC.  Closing.

**Comment by cmeiklejohn** [comment]
Please do not delete this branch, as this code may be retargeted against the 2.1 release.

**Pull request closed without merging**